sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -586,3 +600,295 @@ def finalize(self, metrics):
+        return [DataEntry(data=data_entry)]
+
+
+class MakeSentence(BaseParallelProcessor):
+    """
+    Takes string from text string, makes first symbol uppercae and adds end_symbol if last symbol is alpha.
+    """
+
+    def __init__(
+        self,
+        text_key: str = "text",
+        end_symbol: str = ":",
+        make_uppercase: bool = True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.make_uppercase = make_uppercase
+        self.text_key = text_key
+        self.end_symbol = end_symbol
+
+    def process_dataset_entry(self, data_entry) -> List:
+        if self.make_uppercase:
+            data_entry[self.text_key] = data_entry[self.text_key][0].upper() + data_entry[self.text_key][1:]
+
+        # Check if the last character is not a punctuation and if so, add the end_symbol
+        if data_entry[self.text_key][-1].isalpha():
+            data_entry[self.text_key] += self.end_symbol
+        return [DataEntry(data=data_entry)]
+
+
+class ASRFileCheck(BaseProcessor):
+    def __init__(self, audio_filepath_key: str = "audio_filepath", corrupted_audio_dir: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self.audio_filepath_key = audio_filepath_key
+        self.corrupted_audio_dir = corrupted_audio_dir
+        self.failed_files = []
+
+    def process(self):
+        """Check each file listed in the manifest to ensure it can be loaded with torchaudio."""
+        with open(self.input_manifest_file, 'r') as f:
+            lines = f.readlines()
+
+        entries = []
+        total_lines = len(lines)
+
+        for idx in tqdm(range(total_lines), desc="Checking Audio Files"):
+            line = lines[idx]
+            entry = json.loads(line)
+            try:
+                # Attempt to load the audio file to check if it is corrupted
+                torchaudio.load(entry[self.audio_filepath_key])
+                entries.append(entry)  # File is good, append to entries list
+            except Exception as e:
+                print(f"Failed to load {entry[self.audio_filepath_key]}: {e}")
+                self.failed_files.append(entry[self.audio_filepath_key])  # Log failed file path
+
+                # Move or delete the corrupted audio file
+                if self.corrupted_audio_dir:
+                    # Ensure the directory exists
+                    os.makedirs(self.corrupted_audio_dir, exist_ok=True)
+                    dest_path = os.path.join(
+                        self.corrupted_audio_dir, os.path.basename(entry[self.audio_filepath_key])
+                    )
+                    os.rename(entry[self.audio_filepath_key], dest_path)
+                    print(f"Moved corrupted file to: {dest_path}")
+                else:
+                    os.remove(entry[self.audio_filepath_key])
+                    print(f"Deleted corrupted file: {entry[self.audio_filepath_key]}")
```

Comment:
Can we maybe remove this? I would like to avoid deleting files.
I am worried about what if a user points the processor to a directory containing non-audio files, e.g. their code. Torchaudio will not be able to open them. Code files will get deleted :(

Reply:
Maybe I could add a parameter to solve this issue? The thing is that user should not have code or system files in manifest. 

Maybe I should require user to provide corrupted_audio_dir? In that case we will only move the files, not delete them. The inference time was very long with corrupted files, moreover I was getting errors some times. Since this processor was written for files from toloka, probably there will be corrupted files